### PR TITLE
Dynamically-updating search post fix

### PR DESCRIPTION
**Pull request by bbp9857**
Search posts weren't being updated except for when loading the game.  They now are updated dynamically.

### CODE REVIEW

**Comment by kevlahnota** [comment]
@bbp9857  this was designed passively so it will perform fast on android devices, did you try to test this on entry-tier/mid-tier android since you move the check on draw (the check always run on every draw render). I don't think you check other usages for checkOut() method since it is called elsewhere.

File: forge-gui-mobile/src/forge/adventure/stage/MapSprite.java [hunk 1]
```diff
@@ -53,13 +50,7 @@ public static Array<Actor> GetMapSprites(int chunkX, int chunkY) {
...
         List<Pair<Vector2, Integer>> objects = WorldSave.getCurrentSave().getWorld().GetMapObjects(chunkX, chunkY);
 
-        for (Pair<Vector2, Integer> entry : objects) {
-            BiomeSpriteData data = WorldSave.getCurrentSave().getWorld().getObject(entry.getValue());
-            if (data.layer != SpriteLayer)
-                continue;
-            Actor sprite = new MapSprite(entry.getKey(), WorldSave.getCurrentSave().getWorld().getData().GetBiomeSprites().getSprite(data.name, (int) entry.getKey().x + (int) entry.getKey().y * 11483), null);
-            actorGroup.add(sprite);
-        }
+        actorGroup.addAll(GetMapSpritesBackground(chunkX, chunkY));
```
Comment (kevlahnota) [inline forge-gui-mobile/src/forge/adventure/stage/MapSprite.java:53]: spritelayer is different from background layer, why you remmove this? if you want to remove duplicate code, just update the method to differentiate spritelayer and backgroundlayer

**Comment by bbp9857** [comment]
All of the calls are o(1), but there is admittedly the cost of the additional stack frames.  I did verify that checkOut was unused.

File: forge-gui-mobile/src/forge/adventure/stage/MapSprite.java (hunk 1 above)
Comment (bbp9857) [reply forge-gui-mobile/src/forge/adventure/stage/MapSprite.java:53]: All of the calls are o(1), but there is admittedly the cost of the additional stack frames. I did verify that checkOut was unused.

File: forge-gui-mobile/src/forge/adventure/stage/MapSprite.java [hunk 2]
```diff
@@ -82,9 +73,12 @@ public void draw(Batch batch, float parentAlpha) {
         if (texture == null)
             return;
         batch.draw(texture, getX(), getY());
-        if (isCaveDungeon && !isOldorVisited) {
-            searchPost.setPosition(getX() - 7, getY() + 7);
-            searchPost.draw(batch, parentAlpha);
+        if (isCaveDungeon && point.getData().map != null && point.getID() != null){
```
Comment (kevlahnota) [inline forge-gui-mobile/src/forge/adventure/stage/MapSprite.java:76]: the update should be either on leave or entry of Poi on worldstage not here

File: forge-gui-mobile/src/forge/adventure/stage/MapSprite.java (hunk 1 above)
Comment (kevlahnota) [reply forge-gui-mobile/src/forge/adventure/stage/MapSprite.java:53]: my bad the worldstage is not updated on master but was updated on snapshot version which it is used.. gonna commit it now
image

**Pull request closed without merging**